### ISSUE TITLE
Error: Cannot find module './fetch.node'

### ISSUE THREAD

**Issue by Honour-d-dev** [web3-storage/w3up#1183]
i'm using w3up-client in a Next.js app on the server for delegating capabilities to the frontend users (i.e thats the delegation model from the docs i believe)

and i keep getting this error 
```
Error: Cannot find module './fetch.node'
    at webpackEmptyContext (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\app\page.js:22:10)
    at __webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)     
    at __webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)     
    at __webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)     
    at __webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)     
    at __webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)     
    at eval (./server/w3up-client.ts:11:83)
    at (action-browser)/./server/w3up-client.ts (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\app\page.js:705:1)
    at Function.__webpack_require__ (C:\Users\hp\VsCode\AYA-Project--Blockchain-EHR-System\frontend\.next\server\webpack-runtime.js:33:43)
```

```[tasklist]
### Tasks
```

**Comment by salieflewis** [comment]
I'm running into this error as well.

Unrelated, but are you using w3up inside server actions @Honour-d-dev?

Care to share any code?

**Comment by Honour-d-dev** [comment]
> I'm running into this error as well.
> 
> Unrelated, but are you using w3up inside server actions @Honour-d-dev?
> 
> Care to share any code?

Yeah , i am using it inside a server action and i finally got it working(my bad for not posting a solution)

i installed ipfs-utils then updated my next.config.js file with
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    serverComponentsExternalPackages: ["ipfs-utils"],
  },
};

module.exports = nextConfig;
```
initially though i had to manually resolve the require statement in the `ipfs-utils/src/fetch.js` & `ipfs-utils/src/http/fetch.js`
with 
```js
module.exports = require('native-fetch')
```
&
```js
module.exports = require('./fetch.node')
```
respectively , before finding the next.js configuration.
let me know if anyone works for you

**Comment by salieflewis** [comment]
Also curious if you'd share how you're instantiating a client given most of the example code uses top-level awaits.

Here is what I've written so far.

https://gist.github.com/salieflewis/bbba37daba7acd0edff39ff32e32a72d

**Comment by salieflewis** [comment]
This has been resolved in `@web3-storage/w3up-client: 12.0.0`

@Honour-d-dev

**Comment by travis** [comment]
huge thanks to @gobengo for digging into this!
